--- /etc/init.d/supervisor	2013-06-27 00:01:41.213039032 -0700
+++ supervisor	2013-06-27 00:00:37.137036389 -0700
@@ -26,6 +26,7 @@
 DAEMON=/usr/bin/supervisord
 NAME=supervisord
 DESC=supervisor
+LABEL=$DESC
 
 test -x $DAEMON || exit 0
 
@@ -91,7 +92,7 @@
   start)
 	echo -n "Starting $DESC: "
 	start-stop-daemon --start --quiet --pidfile $PIDFILE \
-		--exec $DAEMON -- $DAEMON_OPTS
+		--startas $DAEMON -- $DAEMON_OPTS
 	test -f $PIDFILE || sleep 1
         if running ; then
             echo "$NAME."
@@ -132,18 +133,17 @@
 	#	just the same as "restart" except that it does nothing if the
 	#   daemon isn't already running.
 	# check wether $DAEMON is running. If so, restart
-	start-stop-daemon --stop --test --quiet --pidfile \
-		/var/run/$NAME.pid --exec $DAEMON \
+	start-stop-daemon --stop --test --quiet --pidfile $PIDFILE\
+		--startas $DAEMON \
 	&& $0 restart \
 	|| exit 0
 	;;
   restart)
     echo -n "Restarting $DESC: "
-	start-stop-daemon --stop --quiet --pidfile \
-		/var/run/$NAME.pid --exec $DAEMON
+	start-stop-daemon --stop --quiet --oknodo --pidfile $PIDFILE
 	[ -n "$DODTIME" ] && sleep $DODTIME
-	start-stop-daemon --start --quiet --pidfile \
-		/var/run/$NAME.pid --exec $DAEMON -- $DAEMON_OPTS
+	start-stop-daemon --start --quiet --pidfile $PIDFILE\
+		--startas $DAEMON -- $DAEMON_OPTS
 	echo "$NAME."
 	;;
   status)
